test/component/BatchFileUploadUpscanNotificationSpec.scala

```diff
@@ -0,0 +1,193 @@
+
+      And("The User Agent header is application/json")
+      requestHeaders.get(USER_AGENT) shouldBe Some("customs-declarations")
+
+      And("The request XML payload contains details of the success outcome")
+      Json.parse(requestPayload) shouldBe Json.parse(expectedFileTransmissionRequest)
+    }
+
+    scenario("Failure request has been made to the Declaration API") {
+      notificationServiceIsRunning()
+      await(repo.create(BatchFileMetadataWithFileOneWithNoCallbackFieldsAndThree)(hasConversationId))
+
+      Given("the Upscan Initiate service has been sent a valid request with this Declaration API as the callback URL")
+
+      When("Upscan Notify service notifies this Declaration API using previously provided callback URL")
+      val validRequest = FakeRequest("POST", endpoint).withJsonBody(FailedJson)
+      val result = route(app = app, validRequest).value
+
+      Then("Declaration Service returns 204")
+      status(result) shouldBe 204
+
+      And("the response body is empty")
+      contentAsString(result) shouldBe empty
+
+      And("a request is made to Custom Notification Service")
+      val (requestHeaders, requestPayload) = eventually(aRequestWasMadeToNotificationService())
+
+      And("The clientSubscriptionId is passed as X-CDS-Client-ID")
+      requestHeaders.get("X-CDS-Client-ID") shouldBe Some(subscriptionFieldsIdString)
+
+      And("The reference is passed as X-Conversation-ID")
+      requestHeaders.get("X-Conversation-ID") shouldBe Some(FileReferenceOne.toString)
+
+      And("The Authorization header contains the value which is configured in the configs")
+      requestHeaders.get(AUTHORIZATION) shouldBe Some(s"Basic $CustomsNotificationAuthHeaderValue")
+
+      And("The Content-Type header is application/xml; charset=UTF-8")
+      requestHeaders.get(CONTENT_TYPE) shouldBe Some("application/xml; charset=UTF-8")
+
+      And("The Accept header is application/xml")
+      requestHeaders.get(ACCEPT) shouldBe Some("application/xml")
+
+      And("The request XML payload contains details of the failure outcome")
+      trim(string2xml(requestPayload)) shouldBe trim(UpscanNotificationFailedCustomsNotificationXml)
+    }
+
+    scenario("Success request has been made to the Declaration API but metadata record does not exist in the database") {
+      notificationServiceIsRunning()
+      await(repo.create(BatchFileMetadataWithFileOneWithNoCallbackFieldsAndThree)(hasConversationId))
+
+      Given("the Upscan Initiate service has been sent a valid request with this Declaration API as the callback URL")
+
+      When("Upscan Notify service notifies this Declaration API using previously provided callback URL")
+      val validRequest = FakeRequest("POST", endpoint).withJsonBody(readyJson(FileReferenceTwo))
+      val result = route(app = app, validRequest).value
+
+      Then("Declaration Service returns 500")
+      status(result) shouldBe 500
+
+      And("the response body is empty")
```

Comment:
`empty` should be replaced with correct response

Reply:
done